tests/pkg_openwsn/udp.c

```diff
@@ -25,32 +27,63 @@
 
 #include "ztimer.h"
 #include "net/ipv6.h"
+#include "net/sock/udp.h"
+#include "od.h"
 
 #include "opendefs.h"
-#include "scheduler.h"
 #include "02a-MAClow/IEEE802154E.h"
 #include "03b-IPv6/icmpv6rpl.h"
-#include "04-TRAN/udp.h"
-#include "cross-layers/openqueue.h"
+
 #include "cross-layers/idmanager.h"
 #include "cross-layers/packetfunctions.h"
 
-extern udp_resource_desc_t uinject_vars;
 extern idmanager_vars_t idmanager_vars;
-extern openudp_vars_t openudp_vars;
 
-static uint16_t counter = 0;
+static sock_udp_t _sock_udp;
+
+void _sock_udp_handler(sock_udp_t *sock, sock_async_flags_t type, void *arg)
+{
+    char buf[50];
+
+    if (type & SOCK_ASYNC_MSG_RECV) {
+        sock_udp_ep_t remote;
+        int16_t res;
+
+        if ((res = sock_udp_recv(sock, buf, sizeof(buf), 0, &remote)) >= 0) {
+            printf("Received %i bytes on port %i\n", res, remote.port);
+            od_hex_dump(buf, res, OD_WIDTH_DEFAULT);
+        }
+    }
+
+    if (type & SOCK_ASYNC_MSG_SENT) {
+        if (*((uint8_t *)arg) == E_FAIL) {
+            puts("Failed to Send");
+        }
+        else {
+            puts("Send Success");
+        }
+    }
+}
+
+void udp_cli_init(void)
+{
+    memset(&_sock_udp, 0, sizeof(sock_udp_t));
+    sock_udp_ep_t local;
 
-OpenQueueEntry_t *pkt;
+    local.family = AF_INET6;
+    local.netif = SOCK_ADDR_ANY_NETIF;
+    local.port = WKP_UDP_ECHO;
 
-void push_pkt_cb(void){
-    owerror_t ret = openudp_send(pkt);
-    if (ret == E_FAIL) {
-        puts("could not send");
-        openqueue_freePacketBuffer(pkt);
+    if (sock_udp_create(&_sock_udp, &local, NULL, 0) < 0) {
+        puts("Could not create socket\n");
+        return;
     }
+
+    sock_udp_set_cb(&_sock_udp, _sock_udp_handler, NULL);
 }
 
+#include "errno.h"
```

Comment:
Should be added at the beginning of the file and use `<>` instead of `""` since it's a toolchain include.

Reply:
addressed